### PR TITLE
Allow empty/undefined for counter min & max

### LINKED ISSUE #8151: Counter Helper GUI doesn't allow for 'Unlimited' setting
## Checklist

- [*] I have updated to the latest available Home Assistant version.
- [*] I have cleared the cache of my browser.
- [*] I have tried a different browser to see if it is related to my browser.

## The problem
Setting up a Counter Helper from the GUI mandates Minimum/Maximum values. There is no way to specify non-integer values (i.e. Unlimited). Accepting the defaults results in a counter that would fail to increment/decrement.

## Expected behavior
One possible solution is to set the some label i.e. 'Leave blank for unlimited'.
102854705-bc65c200-442b-11eb-9cfb-d3494aac2b2a

## Steps to reproduce
1. Setup a new helper entity using the GUI via Configuration->Helpers->Add Helper->Counter
2. Set name and leave all other values as pre-populated defaults (initial value 0, maximum 0, minimum - empty, step - 1)
3. Try to increment it using i.e. Dev Tools->Services->counter.increment using the newly created entity's ID.
4. The counter fails to increment

## Environment

- Home Assistant release with the issue: 2021.1.0
- Last working Home Assistant release (if known): Unknown
- Browser and browser version: Safari 14.01
- Operating system: MacOS X 10.14.6

## State of relevant entities

```yaml

```

## Problem-relevant frontend configuration

```yaml

```

## Javascript errors shown in your browser console/inspector

```txt

```

### PR DESCRIPTION
**Pull request by spacegaier**
## Proposed change

Prevent 0 being set for optional min and max parameters.

## Type of change

- [ ] Dependency upgrade
- [X] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Example configuration

```yaml

```

## Additional information

- This PR fixes or closes issue: fixes https://github.com/home-assistant/frontend/issues/8151
- This PR is related to issue or discussion:
- Link to documentation pull request:

## Checklist

- [X] The code change is tested and works locally.
- [X] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

### CODE REVIEW

File: src/panels/config/helpers/forms/ha-counter-form.ts [hunk 1]
```diff
@@ -44,8 +44,8 @@ class HaCounterForm extends LitElement {
     if (item) {
       this._name = item.name || "";
       this._icon = item.icon || "";
-      this._maximum = item.maximum;
-      this._minimum = item.minimum;
+      this._maximum = item.maximum ?? undefined;
+      this._minimum = item.minimum ?? undefined;
```
Comment (bramkragten) [inline src/panels/config/helpers/forms/ha-counter-form.ts:48]: Why is this needed? If it is undefined, we set it to undefined?

**Review by bramkragten — APPROVED** [review]

File: src/panels/config/helpers/forms/ha-counter-form.ts (hunk 1 above)
Comment (spacegaier) [reply src/panels/config/helpers/forms/ha-counter-form.ts:48]: It comes in as "NaN" but that results in a "0" in the input field. With "undefined" the input stays empty ads intended.
